tests/drivers/at/tests-with-config/test_all_configs.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+handle_urc=0
+echo=0
+rcv_eol_1=""
+rcv_eol_2=""
+send_eol=""
+
+run_test() {
+	echo "Running test with:"
+	echo "  URC handling = $handle_urc"
+	echo "  echo         = $echo"
+	echo "  send EOL     = $send_eol"
+	echo "  rcv EOL 1    = $rcv_eol_1"
+	echo "  rcv EOL 2    = $rcv_eol_2"
+
+	# make -j BOARD=native "HANDLE_URC=$handle_urc" "ECHO_ON=$echo" "SEND_EOL=\"$send_eol\"" "RECV_EOL_1=\"$rcv_eol_1\"" "RECV_EOL_2=\"$rcv_eol_2\"" compile-commands
+	make -j BOARD=native "HANDLE_URC=$handle_urc" "ECHO_ON=$echo" "SEND_EOL=\"$send_eol\"" "RECV_EOL_1=\"$rcv_eol_1\"" "RECV_EOL_2=\"$rcv_eol_2\"" tests-at
+
+	# take /dev/ttyS0 as serial interface. It is only required s.t. UART
+	# initialization succeeds and it gets turned off right away.
+	set +e
+	./bin/native/tests_unittests.elf -c /dev/ttyS0 <<< "s\n"
+	if [ $? -ne 0 ]
```

Comment:
static-tests

Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?. [SC2181]

shellcheck does not like this 

https://www.shellcheck.net/wiki/SC2181

seems like they got good reasons for not liking